notebooks/featurestore/hsfs/time_travel_python.ipynb

```diff
@@ -0,0 +1,1336 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Time travel operations in Hopsworks Feature Store\n",
+    "\n",
+    "In this notebook we will introduce time travel operations in Hopsworks Feature Store (HSFS). Currently HSFS supports Apache Hudi (http://hudi.apache.org/) a storage abstraction/library for doing **incremental** data ingestion to a Hopsworks Feature Store.\n",
+    "\n",
+    "TLDR; Hudi is a storage abstraction/library build on top of Spark. A Hudi dataset stores data in Parquet files and maintains additional metadata to make upserts efficient. A Hudi ingest job is intended to be run as a streaming ingest job, on an interval such as every 15 minutes, reading deltas from a message-bus like Kafka and ingesting the deltas **incrementally** into a data lake.\n",
+    "\n",
+    "![Incremental ETL](./../images/incr_load.png \"Incremetal ETL\")"
```

Comment:
Can we get Lex to make a more modern version of these diagrams?

Reply:
pinged Lex regarding this